packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1424,20 +1380,28 @@ mixin WidgetInspectorService {
     return _followDiagnosticableChain(chain.reversed.toList());
   }
 
-  DiagnosticsSerialisationDelegate _configToDelegate(_SerializeConfig config) {
+  DiagnosticsSerialisationDelegate _createDelegate({
+    String groupName,
+    int subtreeDepth = 1,
+    int maxDescendentsTruncatableNode = -1,
+    bool includeProperties = false,
+    bool expandPropertyValues = true,
+    bool summaryTree = false,
+  }) {
+    final bool interactive = groupName != null;
     final Set<DiagnosticsNode> nodesCreatedByLocalProject = <DiagnosticsNode>{};
     return DiagnosticsSerialisationDelegate(
-      subtreeDepth: config.subtreeDepth,
-      includeProperties: config.includeProperties,
-      expandPropertyValues: config.expandPropertyValues,
+      subtreeDepth: subtreeDepth,
```

Comment:
to me this looks much more like a subclass of DiagnosticsSerializationConfig than a delegate  but maybe I'm missing something.
Not that it really matters but in terms of performance this approach is a lot more costly as you are creating a bunch of closures on every step of traversing the tree.